src/Rule/IsTrue.php

```diff
@@ -28,26 +33,57 @@ final class IsTrue implements RuleWithOptionsInterface, SkipOnErrorInterface, Wh
+     * or bool) are allowed. Defaults to `1` string.
+     * @param bool $strict Whether the comparison to {@see $trueValue} is strict:
+     *
+     * - Strict mode uses `===` operator meaning the type and the value must both match to those set in
+     * {@see $trueValue}.
+     * = Non-strict mode uses `==` operator meaning that type juggling is performed first before the comparison. You can
+     * read more in the PHP docs:
+     *
+     * - https://www.php.net/manual/en/language.operators.comparison.php
+     * - https://www.php.net/manual/en/types.comparisons.php
+     * - https://www.php.net/manual/en/language.types.type-juggling.php
+     *
+     * Defaults to `false` meaning non-strict mode is used.
+     * @param string $messageWithType Error message used when validation fails and neither non-scalar value (int,
+     * float, string, bool) nor null was provided as input. The type is used instead of value here for more predictable
+     * formatting.
+     *
+     * You may use the following placeholders in the message:
+     *
+     * - `{attribute}`: the label of the attribute being validated.
+     * - `{true}` - the value set in {@see $trueValue} option.
+     * - `{type}`: the type of the value being validated.
+     * @param string $messageWithValue Error message used when validation fails and either scalar value (int, float,
+     * string, bool) or null was provided as input.
+     *
+     * You may use the following placeholders in the message:
+     *
+     * - `{attribute}`: the label of the attribute being validated.
+     * - `{true}` - the value set in {@see $trueValue} option.
+     * - `{value}`: the value being validated.
+     * @param bool|callable|null $skipOnEmpty Whether to skip this rule if the validated value is empty / not passed.
+     * See {@see SkipOnEmptyInterface}.
+     * @param bool $skipOnError Whether to skip this rule if any of the previous rules gave an error. See
+     * {@see SkipOnErrorInterface}.
+     * @param Closure|null $when A callable to define a condition for applying the rule. See {@see WhenInterface}.
+     * @psalm-param WhenType $when
+     */
     public function __construct(
-        /**
-         * @var scalar the value representing "true" status. Defaults to `1`.
-         */
         private int|float|string|bool $trueValue = '1',
-        /**
-         * @var bool whether the comparison to {@see $trueValue} is strict. When this is "true", the value and type must
-         * both match {@see $trueValue}. Defaults to "false", meaning only the value needs to be matched.
-         */
         private bool $strict = false,
-        private string $message = 'The value must be "{true}".',
-
-        /**
-         * @var bool|callable|null
-         */
-        private $skipOnEmpty = null,
+        private string $messageWithType = self::DEFAULT_MESSAGE,
+        private string $messageWithValue = self::DEFAULT_MESSAGE,
```

Comment:
Is it really necessary to separate this cases?

Reply:
It's similar to `Boolean` in this regard, so I synced the changes. The reason is predictable formatting of complex structures. The tests were added too.